cmd/govim/internal/fswatcher/os_others.go

```diff
@@ -13,49 +16,157 @@ type fswatcher struct {
 	eventCh chan Event
 	errCh   chan error
 	mw      *fsnotify.Watcher
+	logf    logFn
 }
 
-func New(gomodpath string, tomb *tomb.Tomb) (*FSWatcher, error) {
+func New(root string, filter watchFilterFn, logf logFn, tomb *tomb.Tomb) (*FSWatcher, error) {
 	mw, err := fsnotify.NewWatcher()
 	if err != nil {
 		return nil, fmt.Errorf("failed to create new watcher: %v", err)
 	}
 
+	// activeWatchers is a map keyed by directory paths for directories that has been
+	// added to watch. The bool value is used to dedup multiple remove events for the same
+	// directory from fsnotify (inotify sends two events when removing a dir, "DELETE_SELF"
+	// and "DELETE,ISDIR" where the difference isn't exposed by fsnotify).
```

Comment:
Please can you explain in this comment _how_ the bool does that? i.e. what function does the use of a bool serve? That means you're not jumping below to find the reason :)

Reply:
Added an explanation.